### PR TITLE
Random Battle: Update some Alolan Forme Pokemon

### PR DESCRIPTION
**Pull request by ascriptmaster**
Also add the possibility of Z-Parting Shot.

### CODE REVIEW

**Comment by ascriptmaster** [comment]
Maybe put a chance of adding the crystal? Like how we do with Z-Belly Drum

**Comment by ascriptmaster** [comment]
still waiting for feedback zzzz

**Comment by TheImmortal** [comment]
Persian-A doesn't always get Fur Coat. The Nasty Plot set is fine on Technician.

**Comment by ascriptmaster** [comment]
It also gets literally HP Fighting as its only Technician move. Makes things difficult when you're a Parting Shot set or something but the existence of HP Fighting which is mediocre w/o NP causes it to go Technician randomly.

Also, Persian-A _will_ always get Fur Coat if HP Fighting is removed, making its new set fine as well.

**Comment by ascriptmaster** [comment]
I have updated Alolan Raichu and Ninetales based on their usage stats. I also added a chance for Alolan Raichu to get its Z Crystal for Stoked Sparksurfer, because a nuke move that always paralyzes if the opposition survives is still kinda cool. Also, I reduced the chance of Darkinium Z with Parting Shot, but maybe it should always get the item.

I have reverted Persian's set for now. However, I'm pasting its current usage stats here so you can look at them and figure out what is best for it. The support set that I suggested does seem popular, though, especially with the Darkinium Z for the Z-Parting Shot.

>[23:04] %Ascriptmaster: .usage persianalola moves ou
>[23:04] *TIBot: Parting Shot 90.904% | Foul Play 81.707% | Taunt 81.079% | Toxic 47.388% | Fake Out 20.246% | Dark Pulse 11.567% | Switcheroo 11.390% | Thunderbolt 6.007% | Roar 5.585% | Nasty Plot 5.070% | U-turn 4.978% | Power Gem 4.447% | Snarl 3.292% | Play Rough 2.575% | Hypnosis 2.529% |...
>[23:05] %Ascriptmaster: .usage persianalola items ou
>[23:05] *TIBot: Darkinium Z 63.204% | Leftovers 14.029% | Choice Scarf 10.152% | Focus Sash 2.148% | Life Orb 2.089% | Rocky Helmet 1.734% | Choice Specs 0.973% | Assault Vest 0.832% | Other 4.840%

**Comment by TheImmortal** [comment]
You should probably look at UU, which is similar but still:

```
[17:30] ~The Immortal: .usage persianalola moves uu
[17:30] *TIBot: Parting Shot 92.916% | Foul Play 82.552% | Taunt 79.372% | Toxic 56.033% | Fake Out 17.449% | Dark Pulse 13.614% | Thunderbolt 7.193% | Nasty Plot 6.809% | Switcheroo 6.328% | U-turn 5.465% | Power Gem 4.451% | Snarl 3.850% | Roar 3.546% | Rest 2.211% | Other 18.209%
[17:30] ~The Immortal: .usage persianalola items uu
[17:30] *TIBot: Darkinium Z 60.861% | Leftovers 20.947% | Choice Scarf 5.241% | Rocky Helmet 2.177% | Life Orb 1.976% | Focus Sash 1.088% | Choice Specs 0.974% | Dread Plate 0.834% | Chople Berry 0.677% | Sitrus Berry 0.637% | Other 4.588%
```

**Comment by ascriptmaster** [comment]
Aha, I wasn't sure if UU stats were up yet since the tier is new. Doesn't matter, I no longer have any submitted changes for Persian and as you can see in my last comment I'm just letting you use your judgment on how to work with its move set.

**Comment by ascriptmaster** [comment]
Maybe something like this for Persian since it seems you're waiting for me to fix it even though I said I was going to let you handle it and to accept everything else as is.

    ["nastyplot", "darkpulse", "powergem", "hypnosis", "hiddenpowerfighting", "foulplay", "partingshot", "taunt", "toxic", "snarl"]

Then, reject Snarl with Dark Pulse, and Dark Pulse with Foul Play + no setup, plus Parting Shot with Nasty Plot. Also, force Technician on the specific combination of NP+Snarl, because Snarl has like 2 more base power after Technician boost and that damage is needed if it actually uses an NP set.

**Comment by TheImmortal** [comment]
Sounds good

**Comment by ascriptmaster** [comment]
Implemented everything except the Dark Pulse/Foul Play thing, for the sake of Yveltal and Spiritomb.

**Comment by TheImmortal** [comment]
Hmm, is Taunt necessary? It doesn't care if the opponent sets up because it has Foul Play, Snarl, and Parting Shot.

**Comment by ascriptmaster** [comment]
I looked at the [initial analysis for Alolan Persian on Smogon](http://www.smogon.com/forums/threads/alola-persian.3587536). It seems that Taunt is more useful to hit Status users that can cripple the Pokemon switching in from Parting Shot, moreso than set up sweepers. Also, it's still 79% usage in UU.

**Comment by TheImmortal** [comment]
I'm not 100% sold on the Persian-A. I feel there could be some improvements but I don't know what. If you want to make a separate PR for Persian-A so that the rest of this can be merged then go ahead.

**Comment by ascriptmaster** [comment]
Reverted.

File: data/scripts.js [hunk 1]
```diff
@@ -2166,8 +2163,6 @@ exports.BattleScripts = {
 			item = template.baseStats.spe >= 60 && template.baseStats.spe <= 108 && !counter['priority'] && this.random(3) ? 'Choice Scarf' : 'Choice Specs';
 		} else if (counter.Special >= 3 && hasMove['uturn'] && template.baseStats.spe >= 60 && template.baseStats.spe <= 108 && !counter['priority'] && this.random(3)) {
 			item = 'Choice Scarf';
-		} else if (hasMove['partingshot'] && !teamDetails.zMove && this.random(2)) {
-			item = 'Darkinium Z';
```
Comment (ascriptmaster) [inline data/scripts.js]: Pangoro still has Parting Shot; this change would have affected it and shouldn't have been removed.

Comment (TheImmortal) [inline data/scripts.js]: I'm undecided about that being a good option

**Pull request merged**